shared/auth/generate.go

```diff
@@ -0,0 +1,49 @@
+package auth
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+)
+
+const (
+	// The default length of the API authorization key, in bytes
+	DefaultKeyLength int = 32
```

Comment:
I noticed you're using HMAC-SHA384 above. The [HMAC RFC](https://datatracker.ietf.org/doc/html/rfc2104) states that:

> The definition of HMAC requires a cryptographic hash function, which
   we denote by H, and a secret key K. We assume H to be a cryptographic
   hash function (...)   We denote by B the byte-length of such
   blocks (...) and by L the byte-length of hash outputs (L=16 for MD5, L=20 for
   SHA-1).

> The key for HMAC can be of any length (keys longer than B bytes are
   first hashed using H).  However, less than L bytes is strongly
   discouraged as it would decrease the security strength of the
   function.  Keys longer than L bytes are acceptable but the extra
   length would not significantly increase the function strength. (A
   longer key may be advisable if the randomness of the key is
   considered weak.)

So, tl;dr, you should bump the key size to 384/8 = 48 bytes.

Note: 256-bit key is fine. This comment is so you're protected against crypto-pedants.

Reply:
Done - it now enforces that `GenerateAuthKeyIfNotPresent` will use *at least* 48 bytes. The actual key loader doesn't check the size (other than > 0) but we can enforce that too if it's prudent.